framework/include/base/FEProblem.h

```diff
@@ -1271,5 +1265,36 @@ FEProblem::finalizeUserObjects(const MooseObjectWarehouse<T> & warehouse)
   }
 }
 
+template <class T>
+T &
+FEProblem::getGlobal()
+{
+  // get the class name as map key
+  const std::string name = typeid(T).name();
```

Comment:
`typeid` is pretty consistent these days but we still use the demangle helper in most other places in the framework. We should probably do that here too for consistency. Just grep for "typeid" and you'll see the other uses of this throughout.